enhancements/ignition-spec-dual-support.md

```diff
@@ -0,0 +1,377 @@
+machine is on spec 2 (will happen only once for all existing and new nodes,
+when the MCO with dual support is first deployed onto the cluster). This will
+only be required as part of the MCO.
+
+Note that there exists three types of spec 2 configs:
+ - Those that are directly translatable to spec 3. This is the case for all existing IPI configs.
+ - Those that we’re not 100% sure we can directly translate, but we can infer what the user is doing and do a translation 
+ - Those that we’re 100% sure we CANNOT translate directly, and requires user input for us to correctly translate
+
+During phase 1 we should attempt to translate on a best-effort basis. If the
+cluster can be directly translated to spec 3, we will do so and use the now
+v3 spec version config in the MCD. If not, we will support both, and alert
+the user that there are untranslatable configs.
+
+During phase two, we should fail updates unless the cluster is fully on spec
+3 config. This effectively means that UPI clusters are at risk when upgrading
+to an MCO with dual support. Mitigation methods are discussed below.
+
+For backwards support (spec 3 to spec 2) the configs are 100% translatable.
+
+
+### RHCOS
+
+The RHCOS bootimage needs to be updated to ignition package v2.0+ . Required
+dependencies are discussed below.
+
+
+### Installer
+
+Phase 1:
+
+The installer needs to be updated to generate spec 3 configs for master and
+worker nodes. These are, for now, immediately translated down by the MCS
+when served to spec 2 bootimages. If the user has otherwise defined custom
+machinesets with spec 3 images, those are served spec 3 configs. This means
+that the installer will have to vendor both ignition v0 and v2 until phase 2.
+
+Phase 2:
+
+The bootstrap ignition configs are also updated to serve spec 3, and RHCOS
+images pinned by the installer will be updated to ones with ignition v2 (spec3).
+All spec 2 references are stripped from the installer.
+
+
+### MCO
+
+The MCO and its subcomponents are the most affected by this change. The
+aforementioned spec translator will be housed in the MCO. This means that the
+MCO would need to simultaneously vendor both ignition v0 and v2, and translate
+existing between the spec versions as needed.
+
+Phase 1:
+
+The MCD has the capability to understand both spec 2 and spec 3 configs,
+and lay down files as needed.
+
+The MCC has the capability to translate spec 2 to spec 3 configs. If the
+translation is completely successful, the MCD will be instructed to use
+the spec 3 configs. Otherwise the MCD will continue using the existing
+spec 2 configs, and apply new spec 2/spec 3 configs. If at any point the
+MCC notices it can fully translate to spec 3, that will be the version we
+use.
+
+The MCS will host both spec 3 and spec 2 configs, with a functionality to
+translate spec 3 down to spec 2 configs, and spec 2 up to spec 3 configs.
+The MCS will first check which ignition spec version a new node supports,
+before serving a config.
```

Comment:
is there already something in place that we can leverage? or do we need to check with MAO (or something else?) to provide us with something like that